test/Faker/Provider/InternetTest.php

```diff
@@ -108,6 +108,16 @@ public function testIpv4()
         $this->assertNotFalse(filter_var($this->faker->ipv4(), FILTER_VALIDATE_IP, FILTER_FLAG_IPV4));
     }
 
+    public function testIpv4NotLocalNetwork()
+    {
+        $this->assertNotEquals('1.0.0.0', long2ip(ip2long($this->faker->ipv4()) & (-1 << (32 - 8))));
```

Comment:
> It is not maintainable.

That's a bit exaggerated, but fair enough.
How about ``$this->assertNotRegExp('/\A1\./', $this->faker->ipv4());``
This checks if the IPv4 __starts__ with ``1.``, so ``10.`` does not fail.